lib/jobs/ipmi-job.js

```diff
@@ -127,34 +129,62 @@ function ipmiJobFactory(
             }
             self.addConcurrentRequest(data.host, cmd);
             return ipmiCallBack(data)
-            .then(function(result) {
-                data[cmd] = result;
-                if(data.password) {
-                    delete data.password;
-                }
-                return self._publishIpmiCommandResult(self.routingKey, cmd, data);
-            }).then(function() {
-                return waterline.workitems.findOne({ id: data.workItemId });
-            })
-            .then(function(workitem) {
-                    return waterline.workitems.setSucceeded(null, workitem);
-            })
-            .catch(function (err) {
-                if(data.password) {
-                    delete data.password;
-                }
-                logger.error("Failed to capture IPMI " + cmd +" status data.", {
-                    data: data,
-                    error: err
-                });
-                return waterline.workitems.findOne({ id: data.workItemId })
+                .then(function(result) {
+                    data[cmd] = result;
+                    if(data.password) {
+                        delete data.password;
+                    }
+                    return self._publishIpmiCommandResult(self.routingKey, cmd, data);
+                }).then(function() {
+                    return waterline.workitems.findOne({ id: data.workItemId });
+                })
                 .then(function(workitem) {
-                    return waterline.workitems.setFailed(null, workitem);
+                    var workitemOld = workitem;
+                    return Promise.resolve()
```

Comment:
it could be below to avoid, promise.resolve() and one more level .then.
return waterline.workitems.setSucceeded(null, workitem)
    .then (function () { 
         xxxxx
    });